src/battle/battle_script_commands.c

```diff
@@ -3184,7 +3187,43 @@ BOOL BtlCmd_RapidSpin(void *bw, struct BattleStruct *sp)
         return FALSE;
     }
 
+    //Sticky Web
+    if (sp->side_condition[side] & SIDE_STATUS_STICKY_WEB) {
+        sp->side_condition[side] &= ~SIDE_STATUS_STICKY_WEB;
+        sp->waza_work = MOVE_STICKY_WEB;
+        SkillSequenceGosub(sp, ARC_BATTLE_SUB_SEQ, SUB_SEQ_BLOW_AWAY_HAZARDS_MESSAGE);
+        return FALSE;
+    }
+
+    IncrementBattleScriptPtr(sp, 1);
+
+    return FALSE;
+}
+
+/**
+ * @brief Try to execute the Sticky Web effect.
+ * Stores the user and the users side in sp->scw
+ * Stores the existence of Sticky Web in sp->side_condition to the defenders side
+ * This command will fail if there is Sticky Web already laid on the defenders side.
+ *
+ * @param bw battle work structure
+ * @param sp global battle structure
+ * @return FALSE
+ */
+
+BOOL btl_scr_cmd_FC_trystickyweb(void *bw, struct BattleStruct *sp) {
     IncrementBattleScriptPtr(sp, 1);
+    int adrs = read_battle_script_param(sp);
+    int side = IsClientEnemy(bw, sp->defence_client);
+
+    if (sp->side_condition[side] & SIDE_STATUS_STICKY_WEB) {
+        sp->oneSelfFlag[sp->attack_client].no_pressure_flag = 1;
+        IncrementBattleScriptPtr(sp, adrs);
+    } else {
+        sp->side_condition[side] |= SIDE_STATUS_STICKY_WEB;
+        sp->scw[side].stickyWebBattlerId = sp->attack_client;                      // For Mirror Armor in doubles
```

Comment:
If the Pokémon with this Ability is affected by [Sticky Web](https://bulbapedia.bulbagarden.net/wiki/Sticky_Web_(move)), the effect is reflected back to the Pokémon which set it up. If Pokémon which set up Sticky Web is not on the field, no Pokémon have their Speed lowered.

`sp->attack_client` is the position of the active battlers, would need to somehow handle this in however way you think is good

Reply:
wait so...  mirror armor switch in will put the sticky web back on the other side of the field *IF* the mon that set it up is still there?

Reply:
The stat drop itself, not the hazard.

Reply:
oh thank goodness
so
`sp->scw[side].stickyWebBattlerId` is default 0, client + 1 if active, and we clear it with the other battle flags in [`ClearBattleMonFlags`](https://github.com/BluRosie/hg-engine/blob/3129587bfb4f0b6b3461aaba3f0feb435c17b8dc/src/battle/battle_pokemon.c#L992) maybe?  default `MAX_CLIENT` and `client` if active?

Reply:
Sounds good to me!